mlflow/utils/autologging_utils.py

```diff
@@ -426,6 +426,20 @@ def __call__(self, original, *args, **kwargs):
                 raise e
 
 
+class _AutologgingSessionManager:
+    _session = False
+
+    @classmethod
+    def start_session(cls):
+        cls._session = True
+
+    @classmethod
+    def active_session(cls):
+        return cls._session
```

Comment:
Very simple implementation, but it is unclear that anything beyond acting as an autologging flag is expected of this API from the doc. Lmk if otherwise :)

Reply:
I think we need an `end_session`, else we'll perpetually have an active session after the first time `fit()` is called.

Reply:
Let's make sure to add test coverage for this case (i.e. ensure that the expected exclusivity behavior is exhibited when we call `fit()` multiple times).

Reply:
I agree (I went back and forth on this a lot); but, what if we have some hypothetical integration where it is common to see the code:
```
x()
y()
```
If both `x` and `y` are to be patched, then we'd want both to be captured under the same autologging session. However if we end the autologging session within the patch itself, then every patched function would always begin with no autologging session active - and, depending on how `x` and `y` handle active run state, we may unintentionally not patch `y`.

Reply:
Edit: I am dumb. We could just patch `y` to end the session by calling `end_session`.
I interpreted your comment to mean we would have to begin and end the session within every patch, including `x`